Libraries/src/Amazon.Lambda.AspNetCoreServer.Hosting/Internal/LambdaSnapstartExecuteRequestsBeforeSnapshotHelper.cs

```diff
@@ -0,0 +1,220 @@
+
+/// <summary>
+/// Contains the plumbing to register a user provided <see cref="Func{HttpClient, Task}"/> inside
+/// <see cref="Amazon.Lambda.Core.SnapshotRestore.RegisterBeforeSnapshot"/>.
+/// The function is meant to initialize the asp.net and lambda pipelines during
+/// <see cref="Core.SnapshotRestore.RegisterBeforeSnapshot"/> and improve the
+/// performance gains offered by SnapStart.
+/// <para />
+/// It works by construction a specialized <see cref="HttpClient" /> that will intercept requests
+/// and saved them inside <see cref="LambdaSnapstartInitializerHttpMessageHandler.CapturedHttpRequests" />.
+/// <para /> 
+/// Intercepted requests are then be processed later by <see cref="SnapstartHelperLambdaRequests.ExecuteSnapstartInitRequests"/>
+/// which will route them correctly through a simulated asp.net/lambda pipeline.
+/// </summary>
+internal class LambdaSnapstartExecuteRequestsBeforeSnapshotHelper
+{
+    private readonly LambdaEventSource _lambdaEventSource;
+
+    public LambdaSnapstartExecuteRequestsBeforeSnapshotHelper(LambdaEventSource lambdaEventSource)
+    {
+        _lambdaEventSource = lambdaEventSource;
+    }
+
+    /// <inheritdoc cref="RegisterInitializerRequests"/>
+    [RequiresUnreferencedCode("Serializes object to json")]
+    public void RegisterInitializerRequests(HandlerWrapper handlerWrapper)
+    {
+        #if NET8_0_OR_GREATER
+
+        Amazon.Lambda.Core.SnapshotRestore.RegisterBeforeSnapshot(async () =>
+        {
+            // Construct specialized HttpClient that will intercept requests and saved them inside
+            // LambdaSnapstartInitializerHttpMessageHandler.CapturedHttpRequests.
+            //
+            // They will be processed later by SnapstartHelperLambdaRequests.ExecuteSnapstartInitRequests which will
+            // route them correctly through a simulated lambda pipeline.
+            var messageHandlerThatCollectsRequests = new LambdaSnapstartInitializerHttpMessageHandler(_lambdaEventSource);
+
+            var httpClientThatCollectsRequests = new HttpClient(messageHandlerThatCollectsRequests);
+            httpClientThatCollectsRequests.BaseAddress = LambdaSnapstartInitializerHttpMessageHandler.BaseUri;
+
+            // "Invoke" each registered request function.  Requests will be captured inside.
+            // LambdaSnapstartInitializerHttpMessageHandler.CapturedHttpRequests.
+            await Registrar.Execute(httpClientThatCollectsRequests);
+
+            // Request are now in CapturedHttpRequests.  Serialize each one into a json object
+            // and execute the request through the lambda pipeline (ie handlerWrapper).
+            foreach (var req in LambdaSnapstartInitializerHttpMessageHandler.CapturedHttpRequests)
+            {
+                var json = JsonSerializer.Serialize(req);
+
+                await SnapstartHelperLambdaRequests.ExecuteSnapstartInitRequests(json, times: 5, handlerWrapper);
```

Comment:
Should the `times` be configurable for the user? I assume `5` was picked arbitrary.